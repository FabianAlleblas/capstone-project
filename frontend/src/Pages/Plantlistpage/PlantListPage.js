import { useHistory } from 'react-router-dom'
import styled from 'styled-components/macro'
import PlusButton from '../../Components/Buttons/PlusButton'
import Header from '../../Components/Header/Header'
import PlantListCard from '../../Components/Plantlistcard/PlantListCard'

export default function PlantListPage({ plantList }) {
  const history = useHistory()

  return (
    <>
      <FixedHeader>My Plants</FixedHeader>
      <ListContainer>
        {plantList.map((plant) => (
          <PlantListCard
            key={plant.id}
            plant={plant}
            onClick={() => openDetailPage(plant.id)}
          />
        ))}
      </ListContainer>
      <FixedPlusButton onClick={openAddForm} />
    </>
  )

  function openAddForm() {
    history.push('/addplant')
  }
  function openDetailPage(id) {
    history.push(`/plant?id=${id}`)
  }
}

const FixedHeader = styled(Header)`
  left: 0;
  position: fixed;
  top: 0;
`

const ListContainer = styled.main`
  display: grid;
  gap: 40px 30px;
  grid-template-columns: 1fr 1fr;
  padding: 100px 30px 90px;
`
const FixedPlusButton = styled(PlusButton)`
  bottom: 20px;
  position: fixed;
  right: 20px;
`
