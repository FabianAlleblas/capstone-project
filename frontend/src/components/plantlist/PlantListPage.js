import styled from 'styled-components/macro'
import PlantCard from './PlantCard'

export default function PlantListPage() {
  return (
    <ItemContainer>
      <PlantCard />
    </ItemContainer>
  )
}

const ItemContainer = styled.section``
