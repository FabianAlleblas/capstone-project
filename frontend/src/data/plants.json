[
  {
    "id": 1,
    "name": "Carl",
    "species": "Monstera deliciosa"
  },
  {
    "id": 2,
    "name": "Birgitt",
    "species": "Monstera adansonii"
  },
  {
    "id": 3,
    "name": "Karen",
    "species": "Philodendron gloriosum"
  },
  {
    "id": 4,
    "name": "Jim Jones",
    "species": "Rhaphidophora tetrasperma"
  },
  {
    "id": 5,
    "name": "Samuel",
    "species": "Alocasia zebrina"
  },
  {
    "id": 6,
    "name": "Greg",
    "species": "Scindapsus pictus"
  }
]
