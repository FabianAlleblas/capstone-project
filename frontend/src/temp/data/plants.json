[
  {
    "id": 1,
    "name": "Carl",
    "species": "Monstera deliciosa",
    "specialInfo": "variegated",
    "daysLeft": 4,
    "weeksLeft": 3
  },
  {
    "id": 2,
    "name": "Birgitt",
    "species": "Monstera adansonii",
    "specialInfo": "variegated",
    "daysLeft": 4,
    "weeksLeft": 3
  },
  {
    "id": 3,
    "name": "Karen",
    "species": "Philodendron gloriosum",
    "specialInfo": "variegated",
    "daysLeft": 4,
    "weeksLeft": 3
  },
  {
    "id": 4,
    "name": "Jim Jones",
    "species": "Rhaphidophora tetrasperma",
    "specialInfo": "variegated",
    "daysLeft": 4,
    "weeksLeft": 3
  },
  {
    "id": 5,
    "name": "Samuel",
    "species": "Alocasia zebrina",
    "specialInfo": "variegated",
    "daysLeft": 4,
    "weeksLeft": 3
  },
  {
    "id": 6,
    "name": "Greg",
    "species": "Scindapsus pictus",
    "specialInfo": "variegated",
    "daysLeft": 4,
    "weeksLeft": 3
  }
]
