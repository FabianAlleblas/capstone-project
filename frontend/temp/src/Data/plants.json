[
  {
    "id": 1,
    "name": "Carl",
    "species": "Monstera deliciosa",
    "info": "variegated",
    "daysLeft": 4,
    "weeksLeft": 3
  },
  {
    "id": 2,
    "name": "Birgitt",
    "species": "Monstera adansonii",
    "info": "variegated",
    "daysLeft": 4,
    "weeksLeft": 3
  },
  {
    "id": 3,
    "name": "Karen",
    "species": "Philodendron gloriosum",
    "info": "variegated",
    "daysLeft": 4,
    "weeksLeft": 3
  },
  {
    "id": 4,
    "name": "Jim Jones",
    "species": "Rhaphidophora tetrasperma",
    "info": "variegated",
    "daysLeft": 4,
    "weeksLeft": 3
  },
  {
    "id": 5,
    "name": "Samuel",
    "species": "Alocasia zebrina",
    "info": "variegated",
    "daysLeft": 4,
    "weeksLeft": 3
  },
  {
    "id": 6,
    "name": "Greg",
    "species": "Scindapsus pictus",
    "info": "variegated",
    "daysLeft": 4,
    "weeksLeft": 3
  }
]
