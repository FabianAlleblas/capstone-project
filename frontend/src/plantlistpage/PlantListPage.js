import styled from 'styled-components/macro'
import PlantCard from '../components/plantcard/PlantCard'
import plants from '../data/plants.json'

export default function PlantListPage() {
  return (
    <ItemContainer>
      {plants.map(({ id, name, species }) => (
        <PlantCard key={id} plantname={name} plantspecies={species} />
      ))}
    </ItemContainer>
  )
}

const ItemContainer = styled.section`
  display: grid;
  gap: 40px 20px;
  grid-template-columns: 1fr 1fr;
  padding: 20px;
`
