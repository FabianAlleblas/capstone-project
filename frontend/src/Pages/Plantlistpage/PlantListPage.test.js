import { render } from '@testing-library/react'
import 'jest-styled-components'
import plants from '../../data/plants.json'
import PlantListPage from './PlantListPage'

describe('PlantListPage', () => {
  it('renders correctly', () => {
    const { container } = render(<PlantListPage plantList={plants} />)

    expect(container.firstChild).toMatchSnapshot()
  })

  it('displays the plant names and species', () => {
    const { getByText } = render(<PlantListPage plantList={plants} />)

    plants.forEach(({ plantname }) =>
      expect(getByText(plantname)).toBeInTheDocument()
    )
    plants.forEach(({ plantspecies }) =>
      expect(getByText(plantspecies)).toBeInTheDocument()
    )
  })
})
