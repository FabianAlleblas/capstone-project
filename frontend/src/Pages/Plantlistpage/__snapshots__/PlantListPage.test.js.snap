// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PlantListPage renders correctly 1`] = `
.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  width: 100%;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
  z-index: 10;
}

.c2 {
  -webkit-align-items: baseline;
  -webkit-box-align: baseline;
  -ms-flex-align: baseline;
  align-items: baseline;
  background: var(--primary-dark);
  border-radius: 0 0 50px 0;
  box-shadow: 0 0 4px var(--shadow);
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  padding: 20px 30px 13px 20px;
}

.c2 svg {
  margin-right: 12px;
}

.c3 {
  color: var(--primary-light);
  font-size: 1.75rem;
  font-weight: 700;
  line-height: 1;
}

.c4 {
  margin: 14px 14px 0 0;
}

.c1 {
  left: 0;
  position: fixed;
  top: 0;
}

<header
  class="c0 c1"
>
  <div
    class="c2"
  >
    <h1
      class="c3"
    >
      My Plants
    </h1>
  </div>
  <svg
    class="c4"
  >
    logouticon.svg
  </svg>
</header>
`;
